mm2src/coins/nft/storage/wasm/wasm_storage.rs

```diff
@@ -492,19 +447,18 @@ impl NftTransferHistoryStorageOps for IndexedDbNftStorage {
                 .await?
                 .into_iter()
                 .map(|(_item_id, transfer)| transfer);
-            let filtered = Self::filter_transfers(transfer_tables, filters)?;
+            let filtered = filter_transfers(transfer_tables, filters)?;
```

Comment:
As I understand, we fetch all transfers from IndexedDB and apply filters thereafter. Can cursor be used instead?

Reply:
Yeah, we can build and open cursor and then use  `while let Some((_item_id, item)) = cursor.next()` to get next item and apply filters. But seems like `cursor.next()` doesnt work properly.
I [did such impl](https://github.com/KomodoPlatform/komodo-defi-framework/pull/1914/files#diff-c2d30614cd223699835e416f1071343c867300a53802fee1ab41517910df505aL496-R498) previously in wasm storage method to filter transfers only with empty metadata. However, GUI got runtime error `mm2lib.js:1731 Uncaught Error: closure invoked recursively or after being dropped`. I will send you full logs in dm. 
[PR](https://github.com/KomodoPlatform/komodo-defi-framework/pull/1914) where I started to use cursor `collect` method.

Reply:
Did you look for a way to fix the issue with `next`? Using `collect` or requesting all items from the table will cause excessive memory usage. Also, cursor is used in Zcoin WASM implementation, so we have a possible problem with it too.

Reply:
I tried to use `spawn_local_abortable` function instead of `spawn_local`, but to be honest I dont remember where. Didnt work, I think bcz my idea was wrong. As we needed to do hotfix asap we decided to come back to this issue later.

ps: as I can see now `collect` method utilizes [`next`](https://github.com/KomodoPlatform/komodo-defi-framework/blob/1b10a0666226fb5ff3758020dce2f67c255b9787/mm2src/mm2_db/src/indexed_db/indexed_cursor.rs#L159) and collect works fine.

Reply:
I propose to apply the following plan:
1. Use `next` instead of collect where applicable.
2. Check if `Uncaught Error: closure invoked recursively or after being dropped` happens again.
3. If yes, create an issue in the DeFi framework repo.
4. Troubleshoot and fix the problem.

Reply:
I suppose it can be done in the next iteration :slightly_smiling_face:

Reply:
Thanks for the detailed plan @artemii235 , I totally agree with it :)